nmt_chainer/evaluation.py

```diff
@@ -23,35 +26,57 @@
 def translate_to_file(encdec, eos_idx, test_src_data, mb_size, tgt_indexer, 
                    translations_fn, test_references = None, control_src_fn = None, src_indexer = None, gpu = None, nb_steps = 50,
                    reverse_src = False, reverse_tgt = False,
-                   s_unk_tag = "#S_UNK#", t_unk_tag = "#T_UNK#"):
-    
+                   s_unk_tag = "#S_UNK#", t_unk_tag = "#T_UNK#", postprocess = False):
     log.info("computing translations")
     translations = greedy_batch_translate(encdec, eos_idx, test_src_data, 
                         batch_size = mb_size, gpu = gpu, nb_steps = nb_steps, 
                         reverse_src = reverse_src, reverse_tgt = reverse_tgt)
     
     log.info("writing translation of set to %s"% translations_fn)
     out = codecs.open(translations_fn, "w", encoding = "utf8")
+    if postprocess:
+        ppp_out = codecs.open(translations_fn + ".restored", "w", encoding = "utf8")
     for t in translations:
         if t[-1] == eos_idx:
             t = t[:-1]
 #         out.write(convert_idx_to_string(t, tgt_voc) + "\n")
-        out.write(" ".join(tgt_indexer.deconvert(t, unk_tag = t_unk_tag)) + "\n")
+        deconverted = " ".join(tgt_indexer.deconvert(t, unk_tag = t_unk_tag)) + "\n"
+        out.write(deconverted)
+        if postprocess:
+            ppp_out.write(deconverted.replace("__"+ " ", "").strip() + "\n")
```

Comment:
I do not think the postprocessing should be hard-coded like that. Since you made a ProPostProcessing class, the eval script should reuse this. And should decide on whether to use it  from the config file.